[Mac] Check if vector-based controls enabled before calling style adjustments for vector-based menulist

lilyspiniolas opened the pull request:
#### b96eefd20262e76dd7948b979c563f48e7e9529e
<pre>
[Mac] Check if vector-based controls enabled before calling style adjustments for vector-based menulist
<a href="https://bugs.webkit.org/show_bug.cgi?id=290154">https://bugs.webkit.org/show_bug.cgi?id=290154</a>
<a href="https://rdar.apple.com/147553169">rdar://147553169</a>

Reviewed by Abrar Rahman Protyasha and Aditya Keerthi.

Added check for vector-based control setting in RenderThemeMac::adjustMenuListStyle
before calling RenderThemeCocoa::adjustMenuListStyle.

* Source/WebCore/rendering/mac/RenderThemeMac.mm:
(WebCore::RenderThemeMac::adjustMenuListStyle const):

Canonical link: <a href="https://commits.webkit.org/292504@main">https://commits.webkit.org/292504@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/920af3b0fda6137423511fb30d80197bddf9a7e2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/96225 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/15839 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/5807 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/101290 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/46744 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/98270 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/16135 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/24272 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/73356 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/30590 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/99228 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/12099 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/86930 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/53695 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/11850 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/4691 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/46070 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/81975 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/4788 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/103318 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/23292 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/16962 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/82396 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/23543 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/82949 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/81772 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/26393 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/3833 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/16666 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/15488 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/23255 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/28410 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/22914 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/26394 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/24655 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (approved):
Hmm, unrelated to your change, but maybe we should just say `element` instead of `e`

pxlcoder commented on Source/WebCore/rendering/mac/RenderThemeMac.mm:
```diff
@@ -1015,9 +1015,16 @@ static void setFontFromControlSize(RenderStyle& style, NSControlSize controlSize
-void RenderThemeMac::adjustMenuListStyle(RenderStyle& style, const Element* e) const
+void RenderThemeMac::adjustMenuListStyle(RenderStyle& style, const Element* element) const
 {
-    RenderThemeCocoa::adjustMenuListStyle(style, e);
```
I think there is a subtle behavior change here. Previously, when `vectorBasedControlsOnMacEnabled` is disabled, this would call into `RenderTheme::adjustMenuListStyle`.

We've now removed that call entirely in that path.

pxlcoder commented on Source/WebCore/rendering/mac/RenderThemeMac.mm:
```diff
@@ -1015,9 +1015,16 @@ static void setFontFromControlSize(RenderStyle& style, NSControlSize controlSize
+#else
+    UNUSED_PARAM(element);
+#endif
     NSControlSize controlSize = controlSizeForFont(style);
```
To address the other comment, I think we just need a call to `RenderTheme::adjustMenuListStyle` before this line.

pxlcoder commented on Source/WebCore/rendering/mac/RenderThemeMac.mm:
```diff
@@ -1015,9 +1015,19 @@ static void setFontFromControlSize(RenderStyle& style, NSControlSize controlSize
+        return;
+    }
+#else
+    UNUSED_PARAM(element);
```
This is no longer necessary, `element` is always used.

lilyspiniolas committed:
[Mac] Check if vector-based controls enabled before calling style adjustments for vector-based menulist
https://bugs.webkit.org/show_bug.cgi?id=290154
rdar://147553169

Reviewed by Abrar Rahman Protyasha and Aditya Keerthi.

Added check for vector-based control setting in RenderThemeMac::adjustMenuListStyle
before calling RenderThemeCocoa::adjustMenuListStyle.

* Source/WebCore/rendering/mac/RenderThemeMac.mm:
(WebCore::RenderThemeMac::adjustMenuListStyle const):

Canonical link: https://commits.webkit.org/292504@main
